nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -227,6 +227,13 @@ if [ -z "$_NIXOS_REBUILD_REEXEC" -a -n "$canRun" -a -z "$fast" ]; then
     fi
 fi
 
+# Find configuration.nix and open editor instead of building.
+if [ "$action" = edit ]; then
+    NIXOS_CONFIG=${NIXOS_CONFIG:-$(nix-instantiate --eval --strict -E '<nixos-config>')}
+    exec "${EDITOR:-nano}" "$NIXOS_CONFIG"
+    exit 1
```

Comment:
I'm no bash guru, but I thought exec would relinquish all control, so this line won't ever be executed anyways (and won't be needed), or does it serve any purpose?

Reply:
I've seen this pattern in multiple languages. I'm not sure if there is some weird case that this actually handles or if it's just for clarity.

Reply:
> so this line won't ever be executed anyways 

If the `exec` call itself fails for some reason (e.g. an EDITOR that is not available), the script   would continue and the `exit 1` forces an exit. So in general, this makes sense. But in this particular case it's not necessary because `set -e` at the beginning forces global exit-on-error anyway. But it doesn't hurt either :smile: